src/typings/decoders.ts

```diff
@@ -98,6 +98,15 @@ interface Decoder {
         endIdx: number,
         useFilter: boolean
     ): Nullable<DecodeResult[]>;
+
+    /**
+     * Retrieves the last index of the log event that matches the given timestamp.
+     * If no such log event exists, returns -1.
+     *
+     * @param timestamp
+     * @return
+     */
+    getLogEventIdxByTimestamp(timestamp: number): number;
```

Comment:
Shall we return `null` no such event is found?

Reply:
I just wanna make it consistant between the unstructured & structured `getLogEventIdxByTimestamp`, which return -1  for not found, since their return type is `std::ptrdiff_t`

Reply:
if you think it is fine to make it to null, lmk